nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java

```diff
@@ -468,6 +468,36 @@ private static Root populateElementsRec(
     return result;
   }
 
+  /**
+   * Creates an access path representing a Map get call, where the key is obtained by calling {@code
+   * next()} on some {@code Iterator}. Used to support reasoning about iteration over a map's key
+   * set using an enhanced-for loop.
+   *
+   * @param mapNode Node representing the map
+   * @param iterVar local variable holding the iterator
+   * @param apContext access path context
+   * @return access path representing the get call, or {@code null} if the map node cannot be
+   *     represented with an access path
+   */
+  @Nullable
+  public static AccessPath mapWithIteratorContentsKey(
+      Node mapNode, LocalVariableNode iterVar, AccessPathContext apContext) {
+    List<AccessPathElement> elems = new ArrayList<>();
+    Root root = populateElementsRec(mapNode, elems, apContext);
+    if (root != null) {
+      return new AccessPath(root, elems, new IteratorContentsKey(iterVar.getElement()));
+    }
+    return null;
+  }
+
+  /**
+   * Creates an access path identical to {@code accessPath} (which must represent a map), but with
+   * {@code mapKey} as its map {@code get()} argument
+   */
+  public static AccessPath withMapKey(AccessPath accessPath, MapKey mapKey) {
```

Comment:
So, the way we are using this method is that we have an AP of the form `ap1 = x.y.m.get(iter#numX)` (which is not real, more of a virtual map get) and see `v = iter#numX.next()` so we call this method with `(ap1, v)` to obtain `ap1 = x.y.m.get(v)`. Correct?

If so, maybe this should be `replaceMapKey(...)`? Or, if that's still missing some usage where this is called an access path without an existing map key, then the javadoc should mention it nonetheless "{@code mapKey} as its map {@code get()} argument, replacing any map key information in the original {@code accessPath}."